src/Shared/ThrottlingException.cs

```diff
@@ -0,0 +1,3 @@
+namespace Workleap.DomainEventPropagation;
+
+internal class ThrottlingException() : Exception;
```

Comment:
If at any point this exception can bubble up to the program consuming the library, then this should be public. Otherwise, this is fine.